lib/active_model/serializable_resource.rb

```diff
@@ -1,7 +1,8 @@
 require 'set'
 module ActiveModel
   class SerializableResource
-    ADAPTER_OPTION_KEYS = Set.new([:include, :fields, :adapter])
+    ADAPTER_OPTION_KEYS = Set.new([:include, :fields, :adapter,
+                                   :jsonapi_toplevel_meta])
```

Comment:
I think this should be a global config.  I can't see any reason someone would want to configure any of this from the controller.  Maybe if you're serving different versions of the JSON API standard, but that hasn't happened yet.  There's a PR in the JSON API to add revision at various levels.. But I think that would still be handled via a global `display resource revision` `display document revisions` etc. and the Serializer knows how to get the revision for a given resource.

> A JSON API document MAY include information about its implementation under a top level jsonapi member. 
>  If present, the value of the jsonapi member MUST be an object (a "jsonapi object").
> The jsonapi object MAY contain a version member whose value is a string indicating the highest JSON API version supported. 
>  This object MAY also contain a meta member, whose value is a meta object that contains non-standard meta-information.

Ref: https://github.com/json-api/json-api/pull/824